GitHubClient.py

```diff
@@ -0,0 +1,355 @@
+import os
+import requests
+import json
+
+class GitHubClient(object):
+    """Basic client for getting the last diff and managing issues."""
+    existing_issues = []
+    milestones = []
+
+    def __init__(self):
+        self.github_url = os.getenv('INPUT_GITHUB_URL')
+        if not self.github_url:
+            raise EnvironmentError
+        self.base_url = f'{self.github_url}/'
+        self.repos_url = f'{self.base_url}repos/'
+        self.repo = os.getenv('INPUT_REPO')
+        self.before = os.getenv('INPUT_BEFORE')
+        self.sha = os.getenv('INPUT_SHA')
+        self.commits = json.loads(os.getenv('INPUT_COMMITS')) or []
```

Comment:
Just FYI, but I found that this line raises the following `TypeError` when `INPUT_COMMITS` is undefined.
```
TypeError: the JSON object must be str, bytes or bytearray, not NoneType
```

I didn't want to rely on that error path though, which is why I added the segment above at lines 12-13.

Since it's not relevant to this PR, I didn't include the fix. But if you want to clean this up, this does what you were maybe intending to do:
```python
self.commits = json.loads(os.getenv('INPUT_COMMITS', str([])))
```

Reply:
Never noticed that before, thanks!